easybuild/tools/toolchain/toolchain.py

```diff
@@ -108,6 +116,13 @@ def __init__(self, name=None, version=None, mns=None):
                 self.mod_short_name = self.mns.det_short_module_name(tc_dict)
                 self.init_modpaths = self.mns.det_init_modulepaths(tc_dict)
 
+    def add_class_constants_to_restore(self, names):
+        """Add given constants to list of class constants to restore with each new instance."""
+        if self.CLASS_CONSTANTS_TO_RESTORE is None:
```

Comment:
no, it can't, that's what I had initially... we update `self.CLASS_CONSTANTS_TO_RESTORE`, potentially with many different toolchains in play, so we need a new list instance for each toolchain (not each instance, but each toolchain class, e.g. `Ictce`, `Goolf`, `Intel`, `Foss`, etc.)